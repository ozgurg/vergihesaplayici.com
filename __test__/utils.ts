import type { Component } from "vue";
import type { ComponentMountingOptions } from "@vue/test-utils";
import { mount } from "@vue/test-utils";
import { expect, it } from "vitest";

const TEST_CONTENT = {
    slotHtml: "<p>Default Slot Content</p>",
    attrs: {
        id: "passed-id",
        title: "Passed Title",
        class: "passed-class"
    }
};

export const testDefaultSlot = <T>(component: Component, options?: ComponentMountingOptions<T>): void => {
    it("renders default slot content", () => {
        const wrapper = mount(component, {
            ...options,
            slots: {
                default: TEST_CONTENT.slotHtml
            }
        });
        expect(wrapper.html()).toContain(TEST_CONTENT.slotHtml);
    });
};

export const testAttrPassingToRoot = <T>(component: Component, options?: ComponentMountingOptions<T>): void => {
    it("applies passed attrs to root element", () => {
        const wrapper = mount(component, {
            ...options,
            attrs: TEST_CONTENT.attrs
        });
        expect(wrapper.attributes("id")).toBe(TEST_CONTENT.attrs.id);
        expect(wrapper.attributes("title")).toBe(TEST_CONTENT.attrs.title);
        expect(wrapper.classes()).toContain(TEST_CONTENT.attrs.class);
    });
};

export const testRootClass = <T>(component: Component, rootClass: string, options?: ComponentMountingOptions<T>): void => {
    it("renders with correct root class", () => {
        const wrapper = mount(component, { ...options });
        expect(wrapper.classes()).toContain(rootClass);
    });
};
