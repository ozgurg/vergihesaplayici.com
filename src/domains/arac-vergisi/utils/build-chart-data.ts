import type { CalculationResults, ChartData, Form } from "@/domains/arac-vergisi/types.js";

export const buildChartData = (_: Form, results: CalculationResults): ChartData => {
    const { prices, taxFees } = results;

    const taxRatesData = [];
    taxRatesData.push({
        label: "TRT bandrolü",
        value: taxFees.trt
    });
    taxRatesData.push({
        label: "ÖTV",
        value: taxFees.specialConsumptionTax
    });
    taxRatesData.push({
        label: "KDV",
        value: taxFees.valueAddedTax
    });

    const totalData = [];
    totalData.push({
        label: "Araç",
        value: prices.taxFree
    });
    totalData.push({
        label: "Vergiler",
        value: taxFees.total
    });

    return {
        taxRates: taxRatesData,
        total: totalData
    };
};
