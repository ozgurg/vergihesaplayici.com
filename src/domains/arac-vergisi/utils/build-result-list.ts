import type { CalculationResults, Form, ResultList } from "@/domains/arac-vergisi/types.js";
import { TAX_ADDED_LABEL_BY_MODE, TAX_FREE_LABEL_BY_MODE } from "@/domains/arac-vergisi/config.js";

export const buildResultList = (results: CalculationResults, form: Form): ResultList => {
    const { prices, taxFees, taxRates } = results;
    const list: ResultList = [];

    // üòä
    list.push({
        key: "taxFree",
        label: TAX_FREE_LABEL_BY_MODE[form.mode],
        value: formatMoney(prices.taxFree, "TRY")
    });

    // üòê
    list.push({
        key: "specialConsumptionTax",
        label: `√ñTV (%${taxRates.specialConsumptionTax})`,
        value: formatMoney(taxFees.specialConsumptionTax, "TRY")
    });
    list.push({
        key: "valueAddedTax",
        label: `KDV (%${taxRates.valueAddedTax})`,
        value: formatMoney(taxFees.valueAddedTax, "TRY")
    });

    // üòî
    list.push({
        key: "total",
        label: `Toplam vergi (%${taxRates.total})`,
        value: formatMoney(taxFees.total, "TRY")
    });
    list.push({
        key: "taxAdded",
        label: TAX_ADDED_LABEL_BY_MODE[form.mode],
        value: formatMoney(prices.taxAdded, "TRY")
    });

    return list;
};
