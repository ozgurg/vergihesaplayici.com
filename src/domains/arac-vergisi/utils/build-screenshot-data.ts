import type { Form, ResultList, ScreenshotData } from "@/domains/arac-vergisi/types.js";
import {
    PRICE_LABEL_BY_MODE,
    VEHICLE_ENGINE_VOLUME_LABEL_BY_VEHICLE_ENGINE_VOLUME,
    VEHICLE_ENGINE_VOLUME_LABEL_BY_VEHICLE_TYPE,
    VEHICLE_TYPE_LABEL_BY_VEHICLE_TYPE
} from "@/domains/arac-vergisi/config.js";

export const buildScreenshotData = (form: Form, results: ResultList): ScreenshotData => {
    const input = [];
    input.push({
        label: PRICE_LABEL_BY_MODE[form.mode],
        value: formatMoney(form.price, form.currency),
        isNumeric: true
    });
    input.push({
        label: "Araç türü",
        value: VEHICLE_TYPE_LABEL_BY_VEHICLE_TYPE[form.vehicleType]
    });
    if (form.vehicleEngineVolume) {
        input.push({
            label: VEHICLE_ENGINE_VOLUME_LABEL_BY_VEHICLE_TYPE[form.vehicleType],
            value: VEHICLE_ENGINE_VOLUME_LABEL_BY_VEHICLE_ENGINE_VOLUME[form.vehicleEngineVolume]
        });
    }
    return {
        input: input,
        output: results
    };
};
