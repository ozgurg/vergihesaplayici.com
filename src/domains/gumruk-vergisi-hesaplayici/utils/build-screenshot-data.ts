import type { ResultList, ScreenshotData } from "@/domains/gumruk-vergisi-hesaplayici/types.js";

export const buildScreenshotData = (results: ResultList): ScreenshotData => {
    return {
        input: [],
        output: results
    };
};
