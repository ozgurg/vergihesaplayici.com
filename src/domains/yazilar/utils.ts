import type { Yazi } from "@/domains/yazilar/types.js";
import type { Props as ArticlesGridProps } from "@/components/articles-grid.vue";
import { YazilarSlugPageDef } from "@/domains/yazilar/page-def.js";

export const mapYazilarForArticlesGridComponent = (yazilar: Yazi[]): ArticlesGridProps["items"] => {
    return yazilar.map(_yazi => {
        const yazilarSlugPage = YazilarSlugPageDef({ yazi: _yazi });
        return {
            title: yazilarSlugPage.title,
            description: _yazi.description,
            url: yazilarSlugPage.url,
            date: _yazi.date
        };
    });
};
