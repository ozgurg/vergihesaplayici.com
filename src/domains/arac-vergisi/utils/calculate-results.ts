import type { ExchangeRates } from "@/types/common.js";
import type { CalculationResults, ChartData, Form, ResultList, ScreenshotData } from "@/domains/arac-vergisi/types.js";
import { Calculator } from "@/domains/arac-vergisi/calculator.js";
import { buildResultList } from "@/domains/arac-vergisi/utils/build-result-list.js";
import { buildScreenshotData } from "@/domains/arac-vergisi/utils/build-screenshot-data.js";
import { buildChartData } from "@/domains/arac-vergisi/utils/build-chart-data.js";

export const calculateResults = (params: {
    form: Form,
    exchangeRates: ExchangeRates;
}): {
    results: CalculationResults;
    resultList: ResultList;
    screenshotData: ScreenshotData;
    chartData: ChartData;
} => {
    const { form, exchangeRates } = params;

    const price = form.price * exchangeRates.rates[form.currency];
    const {
        mode,
        vehicleType,
        vehicleEngineVolume
    } = form;

    const calculator = new Calculator({
        price,
        vehicleType,
        vehicleEngineVolume
    }, {
        mode
    });

    const results = calculator.calculate();
    const resultList = buildResultList(form, results);
    const screenshotData = buildScreenshotData(form, resultList);
    const chartData = buildChartData(form, results);

    return {
        results,
        resultList,
        screenshotData,
        chartData
    };
};
