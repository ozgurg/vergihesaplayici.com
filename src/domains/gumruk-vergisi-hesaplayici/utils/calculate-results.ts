import type { ExchangeRates } from "@/types/common.js";
import type {
    CalculationResults,
    Form,
    ResultList,
    ScreenshotData
} from "@/domains/gumruk-vergisi-hesaplayici/types.js";
import { Calculator } from "@/domains/gumruk-vergisi-hesaplayici/calculator.js";
import { buildResultList } from "@/domains/gumruk-vergisi-hesaplayici/utils/build-result-list.js";
import { buildScreenshotData } from "@/domains/gumruk-vergisi-hesaplayici/utils/build-screenshot-data.js";

export const calculateResults = (params: {
    form: Form;
    exchangeRates: ExchangeRates;
}): {
    results: CalculationResults;
    resultList: ResultList;
    screenshotData: ScreenshotData;
} => {
    const { form, exchangeRates } = params;

    const price = form.price * exchangeRates.rates[form.currency];
    const eurToTryCurrency = exchangeRates.rates.EUR;
    const isOverTaxExemptionWeightLimit = form.isOverTaxExemptionWeightLimit;
    const isOverTaxExemptionPriceLimit = form.isOverTaxExemptionPriceLimit;
    const isPrintedOnly = form.isPrintedOnly;
    const isSpecialConsumptionTaxed = form.isSpecialConsumptionTaxed;
    const isFromEU = form.isFromEU;
    const isShippingIncluded = form.isShippingIncluded;
    const extraCustomTaxPercent = form.extraCustomTaxPercent;

    const calculator = new Calculator({
        price,
        eurToTryCurrency,
        isOverTaxExemptionWeightLimit,
        isOverTaxExemptionPriceLimit,
        isPrintedOnly,
        isSpecialConsumptionTaxed,
        isFromEU,
        isShippingIncluded,
        extraCustomTaxPercent
    });

    const results = calculator.calculate();
    const resultList = buildResultList(results, form);
    const screenshotData = buildScreenshotData(resultList);

    return {
        results,
        resultList,
        screenshotData
    };
};
