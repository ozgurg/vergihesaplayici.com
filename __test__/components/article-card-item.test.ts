import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import { testRootClass } from "@root/__test__/utils.js";
import ArticleCardItem from "@/components/article-card-item.vue";

const TEST_ARTICLE = {
    title: "Test Title",
    description: "Test Description",
    url: new URL("https://vergihesaplayici.com"),
    date: new Date("2025-01-01")
};

describe("components/article-card-item.vue", () => {
    testRootClass(ArticleCardItem, "article-card-item", { props: TEST_ARTICLE });

    it("renders card correctly", () => {
        const wrapper = mount(ArticleCardItem, {
            props: TEST_ARTICLE
        });
        expect(wrapper.html({ raw: true })).toMatchSnapshot();
    });
});
