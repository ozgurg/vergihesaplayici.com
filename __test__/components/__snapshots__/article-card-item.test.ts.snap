// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/article-card-item.vue > renders card correctly 1`] = `
"<article data-v-fcd2fd17="" itemscope="" itemtype="https://schema.org/NewsArticle" class="article-card-item"><div data-v-fcd2fd17="" class="meta"><time data-v-fcd2fd17="" datetime="2025-01-01" content="2025-01-01T00:00:00.000Z" itemprop="datePublished" class="meta-date">01.01.2025</time> â€¢ <span data-v-fcd2fd17="" itemscope="" itemprop="author" itemtype="https://schema.org/Person" class="meta-author"><span data-v-fcd2fd17="" itemprop="name">vergihesaplayici.com</span><meta data-v-fcd2fd17="" content="https://vergihesaplayici.com/" itemprop="url"></span></div><h2 data-v-fcd2fd17="" class="title"><a data-v-fcd2fd17="" href="https://vergihesaplayici.com/" itemprop="url" class="stretched-link"><span data-v-fcd2fd17="" itemprop="headline">Test Title</span></a><i data-v-9e7465ed="" data-v-fcd2fd17="" aria-hidden="true" class="svg-icon"><!--
    In \`DEV\` mode, when using \`<use />\` with \`<svg />\`s, Astro fails to load the SVGs properly on the initial load.
    To avoid this issue, I bypass the cache reading in \`DEV\` mode by directly rendering the SVG with \`<svg-fragment />\`.
    --><!--
    <template v-if="isDevMode">
        <svg-fragment :html="props.svg" />
    </template>
    <template v-else>
        <svg>
            <use :href="svgCacheHref" />
        </svg>
    </template>
    --><svg data-v-9e7465ed="" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M13.1717 12.0007L8.22192 7.05093L9.63614 5.63672L16.0001 12.0007L9.63614 18.3646L8.22192 16.9504L13.1717 12.0007Z"></path></svg></i></h2><div data-v-fcd2fd17="" class="summary">Test Description</div></article>"
`;
